src/asynk.rs

```diff
@@ -402,6 +402,25 @@ impl From<crate::Message> for Message {
 }
 
 impl Message {
+
+    /// Creates new empty `Message`, without a Client.
+    /// Useful for passing `Message` data, but cannot be used on it's own for associated methods
+    /// as those require `Client` injected into `Message`.
+    pub fn new() -> Message {
```

Comment:
I wonder if we make the struct items read only but pub, and the constructor here takes subj, reply, msg, headers?

Reply:
I changed the constructor to accept all relevant fields, but  mutability of the struct is a responsibility of the user.

Reply:
The internal members of struct can not be made read only? I thought they could?

Reply:
Unfortunately, no.
We would have to make it private fields and create getter method for it or use some other deref hacks.